admin/users/views.py

```diff
@@ -13,36 +15,104 @@
 
 from admin.base.views import GuidFormView, GuidView
 from admin.users.templatetags.user_extras import reverse_user
+from admin.base.utils import OSFAdmin
+from admin.common_auth.logs import (
+    update_admin_log,
+    USER_2_FACTOR,
+    USER_EMAILED,
+    USER_REMOVED,
+    USER_RESTORED,
+)
 
-from .serializers import serialize_user
-from .forms import EmailResetForm
+from admin.users.serializers import serialize_user
+from admin.users.forms import EmailResetForm
 
 
-def disable_user(request, guid):
-    user = User.load(guid)
-    user.disable_account()
-    user.save()
-    return redirect(reverse_user(guid))
+class UserDeleteView(OSFAdmin, DeleteView):
+    """ Allow authorised admin user to remove/restore user
+
+    Interface with OSF database. No admin models.
+    """
+    template_name = 'users/remove_user.html'
+    context_object_name = 'user'
+    object = None
+
+    def delete(self, request, *args, **kwargs):
+        try:
+            user = self.get_object()
+            flag = None
+            message = None
+            if user.date_disabled is None:
+                user.disable_account()
+                flag = USER_REMOVED
+                message = 'User account {} disabled'.format(user.pk)
+            else:
+                user.date_disabled = None
+                subscribe_on_confirm(user)
+                flag = USER_RESTORED
+                message = 'User account {} reenabled'.format(user.pk)
+            user.save()
+            if flag is not None:
+                update_admin_log(
+                    user_id=self.request.user.id,
+                    object_id=user.pk,
+                    object_repr='User',
+                    message=message,
+                    action_flag=flag
+                )
+        except AttributeError:
+            return page_not_found(
```

Comment:
This doesn't need to happen now, but given how many times this exact `return` happens, it might be nice to add a helper function to do so.